tests/test_quantum_gradients.py

```diff
@@ -452,9 +452,11 @@ def circuit(x, y, z):
         qnode = qml.QNode(circuit, qubit_device_2_wires, diff_method="parameter-shift")
         params = anp.array([0.1, -1.6, np.pi / 5], requires_grad=True)
 
+        h = 1e-4 if qubit_device_2_wires.R_DTYPE is np.float32 else 1e-7
```

Comment:
How come the dtype (I guess precision) is so tightly coupled to the value of `h`?

Reply:
Yes, precisions of the finite difference method depend largely on the data types. 1e-7 just does not makes sense for float32.